### PR TITLE
Ops 665 auth bp and refresh

### PR DESCRIPTION
**Pull request by johndeange**
## What changed

Fixed refresh token/API.
Restructured auth into separate module.

N.B. More auth work to follow.

## Issue

https://github.com/HHS/OPRE-OPS/issues/665

## How to test

Previously it was only possible to be logged in for 30 min.  Now you should be able to stay logged in for 12 hours.

## Definition of Done Checklist
- [x] OESA: Code refactored for clarity
- [x] OESA: Dependency rules followed
- [x] Automated unit tests updated and passed
- [x] Automated integration tests updated and passed
- [x] Automated quality tests updated and passed
- [x] Automated load tests updated and passed
- [x] Automated a11y tests updated and passed
- [x] Automated security tests updated and passed
- [x] 90%+ Code coverage achieved
- ~[ ] Form validations updated~

### CODE REVIEW

File: frontend/src/components/UI/NotificationCenter/NotificationCenter.jsx [hunk 1]
```diff
@@ -1,20 +1,18 @@
...
 import { useDismissNotificationMutation, useGetNotificationsByUserIdQuery } from "../../../api/opsAPI";
-import { jwtDecode } from "jwt-decode";
 import icons from "../../../uswds/img/sprite.svg";
 import customStyles from "./NotificationCenter.module.css";
 import LogItem from "../LogItem";
 import { getAccessToken } from "../../Auth/auth";
+import PropTypes from "prop-types";
 
-const NotificationCenter = () => {
+const NotificationCenter = ({ user }) => {
```
Comment (fpigeonjr) [inline frontend/src/components/UI/NotificationCenter/NotificationCenter.jsx:10]: nice to add some JSDocs for this react component

**Review by fpigeonjr — APPROVED** [review]
💪 ran locally and set a timer for 30 mins...



⭐ I was still logged in.

**Comment by jonnalley** [comment]
> 💪 ran locally and set a timer for 30 mins...
> 
> ⭐ I was still logged in.

@fpigeonjr but were you active in the midst of that 30 minutes?

> Previously it was only possible to be logged in for 30 min. Now you should be able to stay logged in for 12 hours.

@johndeange to be clear, we still need to log you out if you have been idle/inactive after 30 minutes though.  Just want to be sure we're at least not regressing anything in terms of our access/refresh tokens.

**Comment by fpigeonjr** [comment]
> @fpigeonjr but were you active in the midst of that 30 minutes?

No I don't recall being active

**Comment by johndeange** [comment]
@jonnalley  Unfortunately, we have not implemented anything around activity/inactivity - so I wouldn't call it a regression.  We have no user session tracking other than when a user logs in (and maybe logs out).

**Review by devops-chris — APPROVED** [review]

**Review by stevtek — APPROVED** [review]

**Pull request merged**